R/UK.R

```diff
@@ -339,36 +355,31 @@ UK <- R6::R6Class("UK", # rename to country name
         )
         names(self$query_filters) <- self$resolution
       } else {
-        stop(paste("UK data not supported for level", self$level))
+        stop(paste("Uk data not supported for level", self$level))
       }
     },
 
-    #' @description Add NHS data for level 1 regions
+    #' @description Download NHS data for level 1 regions
     #' Separate NHS data is available for "first" admissions, excluding
     #' readmissions. This is available for England + English regions only.
     #'   See: https://www.england.nhs.uk/statistics/statistical-work-areas/covid-19-hospital-activity/ # nolint
     #'     Section 2, "2. Estimated new hospital cases"
+    #' @source https://www.england.nhs.uk/statistics/wp-content/uploads/sites/2/ # nolint
     #' @importFrom lubridate year month
     #' @importFrom readxl read_excel cell_limits
-    #' @importFrom tibble as_tibble
-    #' @importFrom dplyr mutate select %>% group_by summarise left_join
-    #' @importFrom tidyr pivot_longer
-    add_nhs_regions = function() {
+    #' @importFrom dplyr %>%
+    download_nhs_regions = function() {
       if (is.null(self$release_date)) {
         self$release_date <- Sys.Date() - 1
       }
       if (self$release_date < (Sys.Date() - 7)) {
         stop("Data by NHS regions is only available in archived form for the
              last 7 days")
       }
-      message_verbose(
-        self$verbose,
-        "Arranging data by NHS region. Also adding new variable:
+      message("Arranging data by NHS region. Also adding new variable:
               hosp_new_first_admissions. This is NHS data for first hospital
               admissions, which excludes readmissions. This is available for
-              England and English regions only."
-      )
-      # Download NHS xlsx
+              England and English regions only.")
       nhs_url <- paste0(
         "https://www.england.nhs.uk/statistics/wp-content/uploads/sites/2/",
```

Comment:
should this URL be a documented public field?

Reply:
we can put "https://www.england.nhs.uk/statistics/wp-content/uploads/sites/2/" as public field for sure, the issue will be in having the rendered version as it needs to be constructed in a funtion rather than the class body. 

Shall I make a public field called nhs_url which stores the url string and then just sub this in at download?

Reply:
I want to say yes but I am not really sure I know the answer so happy to go with either.

Reply:
Will set base nhs url as a public field and construct the full on in the function.

Reply:
Great to surface this. Would it be too painful to remove the bit at the end so that the public url is "https://www.england.nhs.uk/statistics"? This would make it a usable link which goes somewhere near to the right place. (The remainder of the static url would then just get pasted on inside the function to get the dynamic bit.)